### PR TITLE
Improve maven license detection

### LINKED ISSUE #4262: MIT license not detected in Maven POM
### Description

In this pom.xml:
```XML
<?xml version="1.0" encoding="UTF-8"?>
<project>
    <modelVersion>4.0.0</modelVersion>
    <groupId>com.foo</groupId>
    <artifactId>bar</artifactId>
    <version>1.7.1</version>
    <licenses>
        <license>
            <name>Apache-2.0</name>
        </license>
        <license>
            <name>MIT</name>
        </license>
    </licenses>
</project>
```

The extracted license is:
```YAML
extracted_license_statement: |
    - name: Apache-2.0
    - name: MIT
```

And the detections are:
```YAML
declared_license_expression: apache-2.0
declared_license_expression_spdx: Apache-2.0
license_detections:
    -   license_expression: apache-2.0
        license_expression_spdx: Apache-2.0
        matches:
            -   license_expression: apache-2.0
                license_expression_spdx: Apache-2.0
                from_file: pom.xml
                start_line: 1
                end_line: 1
                matcher: 2-aho
                score: '100.0'
                matched_length: 3
                match_coverage: '100.0'
                rule_relevance: 100
                rule_identifier: spdx_license_id_apache-2.0_for_apache-2.0.RULE
                rule_url: https://github.com/nexB/scancode-toolkit/tree/develop/src/licensedcode/data/rules/spdx_license_id_apache-2.0_for_apache-2.0.RULE
                matched_text: '- name: Apache-2.0'
```

... missing entirely the MIT license

### How To Reproduce

Save a file as pom.xml with content above, then run `scancode --yaml - --package  pom.xml`

### PR DESCRIPTION
**Pull request by pombredanne**
This PR builds on top of:
- https://github.com/aboutcode-org/scancode-toolkit/pull/4215

... to address some issues with Maven license detection including bug fixes.

- This fixes https://github.com/aboutcode-org/scancode-toolkit/issues/4262

- We were only storing the list of attributes but the now "license" top level elements. Because of this a maven tag with only name: mit would not be detected properly as there is no rule for this and this would yield too many false positive. Now we have a structure that is a list of license as {license: {name: mit, url: foo: comments: bar}}

- This PR ensures a more accurate detection. With this also regenerate all Maven related tests, and also ensure we filter early empty and unused license attributes (such as the "distribution")

- Also add new tests and apply mild refactoring for maven license
detection for clarity.

- Remove redundant tests and simplify some other tests for clarity.

### Tasks

* [x] Reviewed [contribution guidelines](https://github.com/nexB/scancode-toolkit/blob/develop/CONTRIBUTING.rst)
* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] Tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR
  Run [tests](https://scancode-toolkit.readthedocs.io/en/latest/contribute/contrib_dev.html#running-tests) locally to check for errors. 
* [x] Commits are in uniquely-named feature branch and has no merge conflicts 📁
* [ ] Updated documentation pages (if applicable)
* [ ] Updated CHANGELOG.rst (if applicable)

### CODE REVIEW

**Comment by pombredanne** [comment]
This can only be merged after:
- https://github.com/aboutcode-org/scancode-toolkit/pull/4215

File: tests/packagedcode/test_plugin_package.py [hunk 1]
```diff
@@ -93,20 +93,6 @@ def test_package_command_scan_npm(self):
...
 
-    def test_package_command_scan_maven_with_license(self):
-        test_dir = self.get_test_loc('maven2')
-        result_file = self.get_temp_file('json')
-        expected_file = self.get_test_loc('plugin/maven-package-with-license-expected.json')
-        run_scan_click(['--package', '--strip-root', '--processes', '-1', test_dir, '--json', result_file])
-        check_json_scan(expected_file, result_file, remove_uuid=True, regen=REGEN_TEST_FIXTURES)
-
-    def test_package_command_scan_npm_with_license(self):
-        test_dir = self.get_test_loc('npm/package')
```
Comment (AyanSinhaMahapatra) [inline tests/packagedcode/test_plugin_package.py:104]: Not sure why the unrelated npm test was removed, makes sense otherwise as the other test files are too large

**Review by AyanSinhaMahapatra — APPROVED** [review]
LGTM! Thanks++ @pombredanne 
This can be merged after we merge https://github.com/aboutcode-org/scancode-toolkit/pull/4215

File: tests/packagedcode/test_plugin_package.py (hunk 1 above)
Comment (pombredanne) [reply tests/packagedcode/test_plugin_package.py:104]: @AyanSinhaMahapatra This test was fully a duplicate of an existing test

Comment (pombredanne) [reply tests/packagedcode/test_plugin_package.py:104]: and it was scanning way too many files at once.
